src/openrct2/scripting/ScriptEngine.cpp

```diff
@@ -1597,44 +1597,48 @@ void ScriptEngine::SetParkStorageFromJSON(std::string_view value)
 
 IntervalHandle ScriptEngine::AllocateHandle()
 {
-    for (size_t i = 0; i < _intervals.size(); i++)
+    auto res = _nextIntervalHandle;
+    _nextIntervalHandle++;
+
+    // In case of overflow start from 1 again
+    if (_nextIntervalHandle == 0)
     {
-        if (!_intervals[i].IsValid())
-        {
-            return static_cast<IntervalHandle>(i + 1);
-        }
+        _nextIntervalHandle = 1;
```

Comment:
Would it work to just do ``
```c++
IntervalHandle ScriptEngine::AllocateHandle()
{
    // In case of overflow start from 1 again
    _nextIntervalHandle = std::max(_nextIntervalHandle++, 1)
    return _nextIntervalHandle;
}
```

Reply:
Good point.